src/core/server/elasticsearch/version_check/ensure_es_version.test.ts

```diff
@@ -192,14 +207,21 @@ describe('pollEsNodesVersion', () => {
 
   it('starts polling immediately and then every esVersionCheckInterval', () => {
     expect.assertions(1);
-    callWithInternalUser.mockReturnValueOnce([createNodes('5.1.0', '5.2.0', '5.0.0')]);
-    callWithInternalUser.mockReturnValueOnce([createNodes('5.1.1', '5.2.0', '5.0.0')]);
+
+    // @ts-expect-error we need to return an incompatible type to use the testScheduler here
+    internalClient.nodes.info.mockReturnValueOnce([
+      { body: createNodes('5.1.0', '5.2.0', '5.0.0') },
+    ]);
+    // @ts-expect-error we need to return an incompatible type to use the testScheduler here
```

Comment:
Will this problem be solved when we switch to a client generated for Kibana?https://github.com/elastic/elasticsearch-js/pull/1239

Reply:
Unfortunately no, this is the 'cannot use promises with marble scheduler' issue were we return an observable or a plain value instead of a promise to be used with `rx.of`. We just did not have the TS error with the legacy client because the mock was not correctly typed.